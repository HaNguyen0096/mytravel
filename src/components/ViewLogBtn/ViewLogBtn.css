@import '~antd/dist/antd.css';

.Btn{
  background-color: orange;
  margin-bottom: 10px;
}
