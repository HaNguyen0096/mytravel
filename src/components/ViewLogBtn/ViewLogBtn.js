import { Button } from 'antd'
import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class ViewLogBtn extends Component {
  render(){
    const {id} = this.props
    const path = `/logs/${id}`
    
    return (
      <div>
        <Button type='primary'><Link to={path}>View More</Link></Button>
      </div>
    )
  }
}