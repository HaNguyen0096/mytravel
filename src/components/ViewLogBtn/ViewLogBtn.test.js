import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ViewLogBtn from './ViewLogBtn'

describe('ViewLogBtn component', () => {

  it('renders ViewLogBtn without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <ViewLogBtn />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for ViewLogBtn as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <ViewLogBtn />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders ViewLogBtn by default', () => {
    const wrapper = shallow(<ViewLogBtn  />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})