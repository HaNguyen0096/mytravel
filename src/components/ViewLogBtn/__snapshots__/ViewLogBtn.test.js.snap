// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewLogBtn component renders ViewLogBtn by default 1`] = `
<div>
  <Button
    block={false}
    className="Btn"
    ghost={false}
    htmlType="button"
    loading={false}
    type="primary"
  >
    <Link
      to="/logs/undefined"
    >
      View More
    </Link>
  </Button>
</div>
`;

exports[`ViewLogBtn component renders the UI for ViewLogBtn as expected 1`] = `
<div>
  <button
    className="ant-btn Btn ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <a
      href="/logs/undefined"
      onClick={[Function]}
    >
      View More
    </a>
  </button>
</div>
`;
